pkg/env/v1alpha1/env.go

```diff
@@ -18,43 +18,141 @@ package v1alpha1
+	// with ENVK. Need to discuss with team.
+
+	// CASTemplateFeatureGateENVK is the ENV key to fetch cas template feature gate
+	// i.e. if cas template based provisioning is enabled or disabled
+	CASTemplateFeatureGateENVK ENVKey = "OPENEBS_IO_CAS_TEMPLATE_FEATURE_GATE"
+
+	// CASTemplateToListVolumeENVK is the ENV key that specifies the CAS Template
+	// to list cas volumes
+	CASTemplateToListVolumeENVK ENVKey = "OPENEBS_IO_CAS_TEMPLATE_TO_LIST_VOLUME"
+
+	// CASTemplateToCreateJivaVolumeENVK is the ENV key that specifies the CAS Template
+	// to create jiva cas volumes
+	CASTemplateToCreateJivaVolumeENVK ENVKey = "OPENEBS_IO_JIVA_CAS_TEMPLATE_TO_CREATE_VOLUME"
+
+	// CASTemplateToReadJivaVolumeENVK is the ENV key that specifies the CAS Template
+	// to read jiva cas volumes
+	CASTemplateToReadJivaVolumeENVK ENVKey = "OPENEBS_IO_JIVA_CAS_TEMPLATE_TO_READ_VOLUME"
+
+	// CASTemplateToDeleteJivaVolumeENVK is the ENV key that specifies the CAS Template
+	// to delete jiva cas volumes
+	CASTemplateToDeleteJivaVolumeENVK ENVKey = "OPENEBS_IO_JIVA_CAS_TEMPLATE_TO_DELETE_VOLUME"
+
+	// CASTemplateToCreateCStorVolumeENVK is the ENV key that specifies the CAS Template
+	// to create cstor cas volumes
+	CASTemplateToCreateCStorVolumeENVK ENVKey = "OPENEBS_IO_CSTOR_CAS_TEMPLATE_TO_CREATE_VOLUME"
+
+	// CASTemplateToReadCStorVolumeENVK is the ENV key that specifies the CAS Template
+	// to read cstor cas volumes
+	CASTemplateToReadCStorVolumeENVK ENVKey = "OPENEBS_IO_CSTOR_CAS_TEMPLATE_TO_READ_VOLUME"
+
+	// CASTemplateToDeleteCStorVolumeENVK is the ENV key that specifies the CAS Template
+	// to delete cstor cas volumes
+	CASTemplateToDeleteCStorVolumeENVK ENVKey = "OPENEBS_IO_CSTOR_CAS_TEMPLATE_TO_DELETE_VOLUME"
+
+	// CASTemplateToCreatePoolENVK is the ENV key that specifies the CAS Template
+	// to create storage pool
+	CASTemplateToCreatePoolENVK ENVKey = "OPENEBS_IO_CAS_TEMPLATE_TO_CREATE_POOL"
+
+	// CASTemplateToDeletePoolENVK is the ENV key that specifies the CAS Template
+	// to delete storage pool
+	CASTemplateToDeletePoolENVK ENVKey = "OPENEBS_IO_CAS_TEMPLATE_TO_DELETE_POOL"
 )
 
+// EnvironmentSetter abstracts setting of environment variable
+type EnvironmentSetter func(envKey ENVKey, value string) (err error)
+
 // EnvironmentGetter abstracts fetching value from an environment variable
-type EnvironmentGetter func(envKey string) (value string)
+type EnvironmentGetter func(envKey ENVKey) (value string)
+
+// EnvironmentLookup abstracts looking up an environment variable
+type EnvironmentLookup func(envKey ENVKey) (value string, present bool)
+
+// Set sets the provided environment variable
+//
+// NOTE:
+//  This is an implementation of EnvironmentSetter
+func Set(envKey ENVKey, value string) (err error) {
+	return os.Setenv(string(envKey), value)
+}
 
 // Get fetches value from the provided environment variable
 //
 // NOTE:
 //  This is an implementation of EnvironmentGetter
-func Get(envKey string) (value string) {
-	return getEnv(envKey)
```

Comment:
Again this has to be generic IMO ....b/c if someone wants to use this helper functions , the key has to be of type ENVKey.